### PR TITLE
samples: sesnor: thermo: add harness_config for testing

### PR DESCRIPTION
**Pull request by hakehuang**
add harness_config for console for testing

### CODE REVIEW

**Comment by hakehuang** [comment]
testing log:

```
~/zephyrproject/zephyr$ ./scripts/twister --device-testing --hardware-map /home/ubuntu/nxp/frdm_mcxn947/map.yaml -p frdm_mcxn947/mcxn947/cpu0 -T samples/sensor/thermometer
ZEPHYR_BASE unset, using "/home/ubuntu/zephyrproject/zephyr"
Renaming output directory to /home/ubuntu/zephyrproject/zephyr/twister-out.1
INFO    - Using Ninja..
INFO    - Zephyr version: v4.0.0-rc1-10-gafce5329f772
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/ubuntu/zephyrproject/zephyr/twister-out/testplan.json

Device testing on:

| Platform                  |         ID | Serial device                                         |
|---------------------------|------------|-------------------------------------------------------|
| frdm_mcxn947/mcxn947/cpu0 | 1063641652 | /dev/serial/by-id/usb-SEGGER_J-Link_001063641652-if00 |

INFO    - JOBS: 8
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  built (not run):    0, skipped:    0, failed:    0, error:    0
INFO    - 1 test scenarios (1 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 1 of 1 test configurations passed (100.00%), 0 built (not run), 0 failed, 0 errored, 0 skipped with 0 warnings in 25.77 seconds
INFO    - In total 1 test cases were executed, 0 skipped on 1 out of total 860 platforms (0.12%)
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.

Hardware distribution summary:

| Board                     |         ID |   Counter |   Failures |
|---------------------------|------------|-----------|------------|
| frdm_mcxn947/mcxn947/cpu0 | 1063641652 |         1 |          0 |
INFO    - Saving reports...
INFO    - Writing JSON report /home/ubuntu/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/ubuntu/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/ubuntu/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed

```

File: samples/sensor/thermometer/sample.yaml [hunk 1]
```diff
@@ -3,11 +3,18 @@ sample:
...
     filter: dt_alias_exists("ambient-temp0")
     integration_platforms:
       - nrf52840dk/nrf52840   # mcp9700a
       - frdm_k22f             # tcn75a
       - robokit1              # ntc_thermistor
       - adi_eval_adin1110ebz  # adt7420
       - frdm_mcxn947/mcxn947/cpu0 # p3t1755
+    harness: console
+    harness_config:
+      fixture: sensor_ambient_temp
```
Comment (MaureenHelm) [inline samples/sensor/thermometer/sample.yaml:15]: Why is a fixture required?

Comment (hakehuang) [reply samples/sensor/thermometer/sample.yaml:15]: if the sensor is not on board but extend port.

**Review by MaureenHelm — APPROVED** [review]

**Review by ubieda — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@MaureenHelm , is this needed for 4.0 release?

**Comment by hakehuang** [comment]
> is this needed for 4.0 release?

@mmahadevan108 , good to have

**Pull request merged**